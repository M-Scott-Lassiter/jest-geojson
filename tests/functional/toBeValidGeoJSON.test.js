const invalidInputValues = [
    undefined,
    null,
    true,
    false,
    200,
    -200,
    Infinity,
    -Infinity,
    NaN,
    [[25, 35, 45000]],
    [
        [
            [0, 0],
            [1, 1]
        ]
    ],
    [[]],
    '',
    'Random Geometry',
    JSON.stringify({
        type: 'LineString',
        coordinates: [
            [25, 90],
            [2, 2]
        ]
    }),
    [[-10, -10, 10, 10]],

    [[-10, -10, 0, 10, 10, 0]],
    [
        [
            {
                type: 'Point',
                coordinates: [100.0, 0.0]
            },
            {
                type: 'LineString',
                coordinates: [
                    [
                        [180.0, 40.0],
                        [180.0, 50.0],
                        [170.0, 50.0],
                        [170.0, 40.0],
                        [180.0, 40.0]
                    ]
                ]
            }
        ]
    ],
    {}
]

// All valid use cases contained within the Geometry, Feature, and FeatureCollection tests.

describe('Invalid Use Cases', () => {
    describe('Expect to fail with bad inputs:', () => {
        test.each(invalidInputValues)('expect(%p).not.toBeValidGeoJSON()', (badInput) => {
            expect(badInput).not.toBeValidGeoJSON()
        })
    })
})

describe('Error Snapshot Testing. Throws error:', () => {
    test(`Valid use case passes`, () => {
        expect(() =>
            expect({ type: 'Point', coordinates: [0, 0] }).not.toBeValidGeoJSON()
        ).toThrowErrorMatchingSnapshot()
    })

    test('Invalid input to matcher', () => {
        expect(() => expect(false).toBeValidGeoJSON()).toThrowErrorMatchingSnapshot()
    })
})
