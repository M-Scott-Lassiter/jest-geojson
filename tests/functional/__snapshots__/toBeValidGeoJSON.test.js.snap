// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Snapshot Testing. Throws error: Invalid input to matcher 1`] = `
"[2mexpect([22m[31mGeoJSONObject[39m[2m).toBeValidGeoJSON()[22m

Argument must be a GeoJSON Geometry, Feature, or FeatureCollection object.

Received:  [31mfalse[39m"
`;

exports[`Error Snapshot Testing. Throws error: Valid use case passes 1`] = `
"[2mexpect([22m[31mGeoJSONObject[39m[2m).not.toBeValidGeoJSON()[22m

Expected input to not be a valid GeoJSON Geometry, Feature, or FeatureCollection object.

Received:  [31m{\\"coordinates\\": [0, 0], \\"type\\": \\"Point\\"}[39m"
`;
